web/src/admin/policies/geoip/GeoIPPolicyForm.ts

```diff
@@ -0,0 +1,170 @@
+        }
+
+        if (data.asnList?.toString() === "") {
+            data.asnList = [];
+        } else {
+            data.asnList = data.asnList.split(",").map(Number);
+        }
+
+        if (data.countryMode?.toString() === "") {
+            data.countryMode = "block";
+        }
+
+        if (data.countryList?.toString() === "") {
+            data.countryList = [];
+        } else {
+            data.countryList = data.countryList.split(",");
+        }
+
+        if (this.instance) {
+            return new PoliciesApi(DEFAULT_CONFIG).policiesGeoipUpdate({
+                policyUuid: this.instance.pk || "",
+                geoIPPolicyRequest: data,
+            });
+        } else {
+            return new PoliciesApi(DEFAULT_CONFIG).policiesGeoipCreate({
+                geoIPPolicyRequest: data,
+            });
+        }
+    }
+
+    renderForm(): TemplateResult {
+        return html` <span>
+                ${msg(
+                    "Ensure the user satisfies requirements of geography or network topology, based on IP address.",
+                )}
+            </span>
+            <ak-form-element-horizontal label=${msg("Name")} ?required=${true} name="name">
+                <input
+                    type="text"
+                    value="${ifDefined(this.instance?.name || "")}"
+                    class="pf-c-form-control"
+                    required
+                />
+            </ak-form-element-horizontal>
+            <ak-form-element-horizontal name="executionLogging">
+                <label class="pf-c-switch">
+                    <input
+                        class="pf-c-switch__input"
+                        type="checkbox"
+                        ?checked=${first(this.instance?.executionLogging, false)}
+                    />
+                    <span class="pf-c-switch__toggle">
+                        <span class="pf-c-switch__toggle-icon">
+                            <i class="fas fa-check" aria-hidden="true"></i>
+                        </span>
+                    </span>
+                    <span class="pf-c-switch__label">${msg("Execution logging")}</span>
+                </label>
+                <p class="pf-c-form__helper-text">
+                    ${msg(
+                        "When this option is enabled, all executions of this policy will be logged. By default, only execution errors are logged.",
+                    )}
+                </p>
+            </ak-form-element-horizontal>
+            <ak-form-group .expanded=${true}>
+                <span slot="header"> ${msg("Policy-specific settings")} </span>
+                <div slot="body" class="pf-c-form">
+                    <ak-form-element-horizontal label=${msg("ASN Mode")} name="asnMode">
+                        <ak-search-select
+                            .fetchObjects=${async () => {
+                                return [
+                                    { name: "allow", label: "Allow" },
```

Comment:
the value for this (regardless of which UI component is used) should use the enum from the API instead of the raw string